pkg/tsdb/cloudwatch/models/logs_query.go

```diff
@@ -1,28 +1,18 @@
 package models
 
-type LogGroup struct {
-	ARN       string `json:"arn"`
-	Name      string `json:"name"`
-	AccountID string `json:"accountId"`
-}
+import (
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/kinds/dataquery"
+)
 
 type LogsQuery struct {
-	LogType            string `json:"type"`
-	SubType            string
-	Limit              *int64
-	Time               int64
-	StartTime          *int64
-	EndTime            *int64
-	LogGroupName       string
-	LogGroupNames      []string
-	LogGroups          []LogGroup `json:"logGroups"`
-	LogGroupNamePrefix string
-	LogStreamName      string
-	StartFromHead      bool
-	Region             string
-	QueryString        string
-	QueryId            string
-	StatsGroups        []string
-	Subtype            string
-	Expression         string
+	dataquery.CloudWatchLogsQuery
+	StartTime     *int64
+	EndTime       *int64
+	Limit         *int64
+	LogGroupName  string
+	LogStreamName string
+	QueryId       string
+	QueryString   string
+	StartFromHead bool
+	Subtype       string
```

Comment:
sort of like my Type question, would you be able to explain why these extra fields aren't part of the generated type? It's because it's not part of the frontend...?

Reply:
from the docs on how to write composable kinds:

> Remember that schemas must describe only persisted data ... For dataquery composable kinds, following the “only persisted” rule generally means omitting fields from schema that are set only at runtime for a single execution of a particular query.

so these fields are being added by the frontend when executing the query.